R/layer.r

```diff
@@ -229,6 +229,15 @@ Layer <- ggproto("Layer", NULL,
     evaled <- lapply(aesthetics, rlang::eval_tidy, data = data)
     evaled <- compact(evaled)
 
+    nondata_cols <- check_nondata_cols(evaled)
+    if(length(nondata_cols) > 0){
+      msg <- paste0(
+        "Aesthetics must be valid data columns: ", nondata_cols,
+        ". Did you forget to add stat()?"
```

Comment:
Is forgetting `stat()` really the most likely cause?

Reply:
There's two ways to get to this point: Forgetting `stat()` or mistyping a column name in such way that the result is a function, e.g. typing `mean` when the data column is called `Mean`. How about: `"Did you mistype the name of a data column or forget to add stat()?"`

Reply:
Can we display the thing that they actually typed? Then we could hint to check that `blah` is a column in the data frame.

Reply:
I don't know how to get it to print the actual typed statement, but now it prints the problematic aesthetic (along with the tilde). 

Situation 1: 
```ggplot(mtcars, aes(x = mpg, stat(density))) + geom_tile(stat = "density") + geom_point()```

Error: Aesthetics must be valid computed stats: `~stat(density)`. Did you map your stat in the wrong layer? 

Situation 2:
```
ggplot(mtcars, aes(x = mpg, fill = density)) + geom_tile(stat = "density") + geom_point()
```
Don't know how to automatically pick scale for object of type function. Defaulting to continuous.
 Error: Aesthetics must be valid data columns: `~density`. Did you mistype the name of a data column or forget to add stat()?